lib/rides/widgets/search_suggestion_filter.dart

```diff
@@ -0,0 +1,439 @@
+      Widget deviationWidget = Row(
+        children: [const Icon(Icons.schedule), const SizedBox(width: 6), Text("± ${_maxDeviationController.text}")],
+      );
+      widgets.add(deviationWidget);
+    }
+    Widget sortingWidget = Row(
+      children: [const Icon(Icons.sort), Text(_sorting.getDescription(context))],
+    );
+    widgets.add(sortingWidget);
+    int numDividers = widgets.length - 1;
+    for (int i = 0; i < numDividers; i++) {
+      widgets.insert(i * 2 + 1, const VerticalDivider(indent: 5, endIndent: 5));
+    }
+    return IntrinsicHeight(
+      child: Row(
+        children: [
+          IconButton(
+            onPressed: () => dialog(context, setState),
+            icon: const Icon(Icons.tune),
+            tooltip: S.of(context).pageSearchSuggestionsTooltipFilter,
+          ),
+          Expanded(
+            child: Center(
+              child: SingleChildScrollView(
+                scrollDirection: Axis.horizontal,
+                child: Row(
+                  children: widgets,
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+
+  List<Ride> apply(List<Ride> rideSuggestions, DateTime date) {
+    return rideSuggestions
+        .where(
+          (Ride ride) {
+            Profile driver = ride.drive!.driver!;
+            bool ratingSatisfied = driver.getAggregateReview().rating >= _minRating &&
+                driver.getAggregateReview().comfortRating >= _minComfortRating &&
+                driver.getAggregateReview().safetyRating >= _minSafetyRating &&
+                driver.getAggregateReview().reliabilityRating >= _minReliabilityRating &&
+                driver.getAggregateReview().hospitalityRating >= _minHospitalityRating;
+            bool featuresSatisfied = Set.of(driver.profileFeatures!).containsAll(_selectedFeatures);
+            bool maxDeviationSatisfied =
+                date.difference(ride.startTime) < Duration(hours: int.parse(_maxDeviationController.text));
+            return ratingSatisfied && featuresSatisfied && maxDeviationSatisfied;
+          },
+        )
+        .sorted(_sorting.sortFunction(date))
+        .toList();
+  }
+
+  void dispose() {
+    _maxDeviationController.dispose();
+  }
+}
+
+enum SearchSuggestionSorting { relevance, time, price }
+
+extension SearchSuggestionSortingExtension on SearchSuggestionSorting {
+  String getDescription(BuildContext context) {
+    switch (this) {
+      case SearchSuggestionSorting.relevance:
+        return S.of(context).searchSuggestionsSortingRelevance;
+      case SearchSuggestionSorting.time:
+        return S.of(context).searchSuggestionsSortingTime;
+      case SearchSuggestionSorting.price:
+        return S.of(context).searchSuggestionsSortingPrice;
+    }
+  }
+
+  int Function(Ride, Ride) sortFunction(DateTime date) {
+    timeFunc(Ride ride1, Ride ride2) => (date.difference(ride1.startTime) - date.difference(ride2.startTime)).inMinutes;
```

Comment:
we could declare those inside the extension so they don't have to be redefined every time. But now that I have seen my own code for that, I don't want it anymore :(


```
  getTimeFunc(date) => (Ride ride1, Ride ride2, DateTime date) =>
      (date.difference(ride1.startTime) - date.difference(ride2.startTime)).inMinutes;
  int priceFunc(Ride ride1, Ride ride2) => ((ride1.price! - ride2.price!) * 100).toInt();

  int Function(Ride, Ride) sortFunction(DateTime date) {
    switch (this) {
      case SearchSuggestionSorting.relevance:
        return (ride1, ride2) => getTimeFunc(date)(ride1, ride2) + priceFunc(ride1, ride2);
      case SearchSuggestionSorting.time:
        return getTimeFunc(date);
      case SearchSuggestionSorting.price:
        return priceFunc;
    }
  }
```

Reply:
So you're taking back your own proposal? I'm confused :D I think since you need the date for the timeFunc, it has to be redefined every time regardless, and it would be kind of weird to have those at different places or have one built by a function with an argument, as you proposed. I think performance isn't that much of an issue that we have to save this tiny function declaration (and if the Dart compiler is smart, it saves it anyway)

Reply:
Yes but I did not want to have my work got to waste so I just posted how you _would_ do it :D